ojdbc-provider-jackson-oson/src/main/java/oracle/jdbc/provider/oson/deser/OsonBigIntegerDeserializer.java

```diff
@@ -0,0 +1,99 @@
+ ** either these or other terms.
+ **
+ ** This license is subject to the following condition:
+ ** The above copyright notice and either this complete permission notice or at
+ ** a minimum a reference to the UPL must be included in all copies or
+ ** substantial portions of the Software.
+ **
+ ** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ ** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ ** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ ** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ ** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ ** OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ ** SOFTWARE.
+ */
+
+package oracle.jdbc.provider.oson.deser;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.NumberDeserializers;
+import com.fasterxml.jackson.databind.deser.std.StdScalarDeserializer;
+
+import com.fasterxml.jackson.databind.node.TreeTraversingParser;
+import oracle.jdbc.provider.oson.OsonParser;
+
+import java.io.IOException;
+import java.math.BigInteger;
+
+/**
+ * Deserializer class for handling BigInteger objects using {@link OsonParser}.
+ * This class extends {@link StdScalarDeserializer} to enable the deserialization of BigInteger
+ * values from JSON data.
+ * <p>
+ * It overrides the {@link #deserialize(JsonParser, DeserializationContext)} method to provide
+ * a custom implementation that uses the Oson library's {@link OsonParser#getBigIntegerValue()} method
+ * for extracting BigInteger values from the JSON content.
+ * </p>
+ *
+ * @see StdScalarDeserializer
+ * @see OsonParser
+ * @see BigInteger
+ */
+public class OsonBigIntegerDeserializer extends StdScalarDeserializer<BigInteger> {
+
+  /**
+   * A singleton instance of the deserializer.
+   */
+  public static final OsonBigIntegerDeserializer INSTANCE = new OsonBigIntegerDeserializer();
+
+  /**
+   * Default constructor that initializes the deserializer for the {@link BigInteger} class.
+   */
+  protected OsonBigIntegerDeserializer() {
+    super(BigInteger.class);
+  }
+
+  /**
+   * Deserializes a BigInteger from the JSON input using the {@link OsonParser}.
+   *
+   * @param p the {@link JsonParser} for reading the JSON content
+   * @param ctxt the deserialization context
+   * @return the deserialized {@link BigInteger} value
+   * @throws IOException if there is a problem with reading the input
+   */
+  @Override
+  public BigInteger deserialize(JsonParser p, DeserializationContext ctxt) throws IOException{
+    if( p instanceof TreeTraversingParser) {
+      return NumberDeserializers.BigIntegerDeserializer.instance.deserialize(p, ctxt);
+    }
+    else {
+      final OsonParser _parser = (OsonParser)p;
```

Comment:
Are we sure that TreeTraversingParser is the only other type of parser here? Same comment for all other deserializers.

Reply:
It's being created on using the ObjectNode using object mapper. For the rest of the parsers, we override it.